CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -554,7 +554,7 @@ public static function buildRecur(&$form) {
     $form->add('checkbox', 'is_recur', ts('I want to contribute this amount'), NULL);
 
     if (!empty($form->_values['is_recur_interval']) || $className == 'CRM_Contribute_Form_Contribution') {
-      $form->add('text', 'frequency_interval', ts('Every'), $attributes['frequency_interval']);
```

Comment:
@eileenmcnaughton I thought that would be too sweeping a change, and prefer localized fixes to the forms affected.

Reply:
hmm ok - it feels like adding an aria-class is kinda a 'safe' change though - would only affect accessibility users, who currently have no accessibility to break?

Reply:
Yes exactly, it adds an attribute aria-label which make that form element accessible by label and doing so it doesn't break anything. I drop my patch which adds this additional attribute implicitly when a title is present because it's not required for each case and thus it is better to manually pick such element and add this attribute which I did in this PR

Reply:
What cases is it not required for?

Reply:
Case when label is rendered against from element in tpl:
```php
{$form.element.label} {$form.element.html}
```
Thus compiled into 
<label for="element_id"> <input type="text" id="element_id" />
Here the ```for``` attribute is equivalent to ```aria-labelledby``` that indicate this label belongs to which form element. For those cases where label is not used or declared backend, we need to declare the aria-label manually as I did in this patch.

Reply:
hmm - but that still seems like a pattern that could be handled when adding the label in QF?

Reply:
In a case when user don't want to declare label (```{$form.element.label}```) in tpl but to make that element accessible by label, we should then need to add aria-label attribute in QF. In such case I found no pattern where we can't expect label to be declared always against its form element.

Reply:
Hmm OK - so you are saying some fields don't have a label so in those cases we need to still display aria so

 $frequencyUnit = $form->addElement('select', 'frequency_unit', NULL, $units);

for this there is no label

& we need to pass in an aria label if we want one but in this case

$form->add('text', 'frequency_interval', ts('Every'), $attributes['frequency_interval']);

We have a label so we could use that for aria without double- defining it couldn't we?

Reply:
@eileenmcnaughton my argument against automatically rendering an `aria-label` attribute for every form element is that it makes a ton of extra markup and doesn't actually solve the problem. Most of the time when we don't print a label in the tpl, the quickform element has a NULL label as well, so our clever auto-fix doesn't actually fix anything. Since a manual audit is needed anyway, we might as well skip the auto-fix cleverness.

Reply:
Hmm - but in the medium term we are ideally using the labels from the QF elements more (& ideally from addField) so setting a label in the QF add & using {$form.element.label}) seems better practice?

I guess if we were using the new fieldsArray protocol we could add when we add the fields from there too.

Reply:
Well the standard "auto form tpl" that you are moving us to will always have elements printed with their labels, so it's only in odd tpl override situations where we'd need it.